platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -1127,18 +1477,115 @@ namespace platoon_strategic_ihp
             }
         }
         
-        else {
-            ROS_DEBUG_STREAM("Received mobility request with type " << msg.plan_type.type << " and ignored.");
-            return MobilityRequestResponse::NO_RESPONSE;
-        }
-    }
-    
-    // UCLA: mobility request leader aborting (inherited from candidate follower)
-    MobilityRequestResponse PlatoonStrategicIHPPlugin::mob_req_cb_leaderaborting(const cav_msgs::MobilityRequest& msg)
-    {
-        // This state does not handle any mobility request for now
-        // TODO Maybe it should handle some ABORT request from a waiting leader
-        ROS_DEBUG_STREAM("Recived mobility request with type " << msg.plan_type.type << " but ignored.");
+        // UCLA: conditions for cut-in join; platoon leader --> leading with operation
+        else if (isCutinJoin)
+        {
+            /* We are currently checking two basic JOIN conditions:
+             *     1. The size limitation on current platoon based on the plugin's parameters.
+             *     2. Calculate how far away the joining vehicle to the platoon (longitudinal & lateral),  
+             *        Determine if it was in a reasonable distance for cut-in join.
+             */ 
+            
+            cav_msgs::MobilityHeader msgHeader = msg.header;
```

Comment:
This block of ~40 lines beginning here is repeated in each of these 3 major if/else blocks.  It should be pulled out and executed above the if statement at the top of the method, thus substantially shrinking the code size and making it more maintainable.